specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json

```diff
@@ -4286,15 +4286,55 @@
           "description": "Specifies whether to verify the identity of the server when connecting over SSL. The default value is true. Type: boolean (or Expression with resultType boolean).",
           "type": "object"
         },
-        "encryptedCredential": { 
-          "description": "The encrypted credential used for authentication. Credentials are encrypted using the integration runtime credential manager. Type: string (or Expression with resultType string).", 
+        "encryptedCredential": {
+          "description": "The encrypted credential used for authentication. Credentials are encrypted using the integration runtime credential manager. Type: string (or Expression with resultType string).",
           "type": "object"
         }
       },
       "required": [
         "endpoint",
         "clientId"
-      ] 
+      ]
+    },
+    "AzureFunctionLinkedService": {
+      "x-ms-discriminator-value": "AzureFunction",
+      "description": "Azure Function linked service.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/LinkedService"
+        }
+      ],
+      "properties": {
+        "typeProperties": {
+          "description": "Azure Function linked service properties.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AzureFunctionLinkedServiceTypeProperties"
+        }
+      },
+      "required": [
+        "typeProperties"
+      ]
+    },
+    "AzureFunctionLinkedServiceTypeProperties": {
+      "description": "Azure Function linked service properties.",
+      "properties": {
+        "functionAppUrl": {
+          "description": "The endpoint of the Azure Function App. URL will be in the format https://<accountName>.azurewebsites.net.",
+          "type": "object"
+        },
+        "functionKey": {
+          "description": "Function or Host key for Azure Function App.",
+          "type": "object"
+        },
+        "encryptedCredential": {
+          "description": "The encrypted credential used for authentication. Credentials are encrypted using the integration runtime credential manager. Type: string (or Expression with resultType string).",
+          "type": "object"
+        }
```

Comment:
The type of functionAppUrl & functionKey should be `string`, right? (instead of object)

Reply:
functionKey is a secret so we support multiple ways to refer to it (e.g. AzureKeyVaultSecretReference or SecretBase) and hence its an object. Currently functionAppUrl is treated as a string but we might want to allow parameters for it in the future, since customers might want to change the value dynamically while deploying to different environments. 
Having it as object here since it will be a breaking change to change from type string to object, and we will likely support it in the future

Reply:
Ok, I see. Can we update the description of functionKey to make it clear that it's value can be an expression and well? (same comment for functionName)

Reply:
Updated functionName for now, functionKey support for KeyVaultReference is still in progress, so will update the description once it is ready. It will be non-breaking since the type is already supported

Reply:
@anuchandy can we complete this PR now. Ill update the description for functionKey in about 2-3 weeks once keyVault support is fully rolled out

Reply:
Functionality is now working and tested so im making the appropriate changes for functionKey too